Azure/azure-cli
# {App Service} Bugfix: az webapp create error handling, appservice tests

**Description**  
- Improve error handling for `az webapp create`
- Test improvements so test_webapp_commands file can run live. 

Fixes #13488

**Testing Guide**  
- `az webapp create -g RG --plan NonExistentPlan --name "WebappName"` Should return error "The plan {} doesn't exist in the resource group {} instead of "error: nonetype has no attribute location"
- `az webapp create -g RG --plan Plan --name "Invalid***name"` Should return "Hostname must be less than 64 chars and must follow characte rguidelines..." instead of "The resource Microsoft.web/sites/Iinvalid***name under resource group {} was not found..."
- `az webapp create --resource-group RG --plan Plan --name ExistingWebappNameFromDiffRG` Should return "The webapp {} exists in resource group {} and does not match the value entered {}. Please re-run command" instead of "The resource Microsoft.web/sites/Iinvalid***name under resource group {} was not found..."

- `azdev test test_webapp_commands --live` should pass


---

This checklist is used to make sure that common guidelines for a pull request are followed.
